lib/ansible/modules/storage/netapp/na_ontap_command.py

```diff
@@ -119,6 +149,64 @@ def apply(self):
         output = self.run_command()
         self.module.exit_json(changed=changed, msg=output)
 
+    def parse_xml_to_dict(self, xmldata):
+        '''Parse raw XML from system-cli and create an Ansible parseable dictonary'''
+        try:
```

Comment:
I would prefer to have smaller try blocks.  EG
```
try:
     import xml.parsers.expat
except xml.parsers.expat.ExpatError as errcode:
     self.result_dict['status'] = "XML parser: " + str(errcode)
     return self.result_dict
```

Same thing with:
```
   try:
         xml_parser.Parse(xml_str)
```

but since you're checking specific exceptions, I don't think it matters too much.

Reply:
What about something like this?
```python
    def parse_xml_to_dict(self, xmldata):
        '''Parse raw XML from system-cli and create an Ansible parseable dictonary'''
        xml_import_ok = True
        xml_parse_ok = True

        try:
            import xml.parsers.expat
        except ImportError:
            self.result_dict['status'] = "XML parsing failed. Cannot import xml.parsers.expat!"
            self.result_dict['stdout'] = str(xmldata)
            xml_import_ok = False

        if xml_import_ok:
            xml_str = xmldata.decode('utf-8').replace('\n', '---')
            xml_parser = xml.parsers.expat.ParserCreate()
            xml_parser.StartElementHandler = self._start_element
            xml_parser.CharacterDataHandler = self._char_data
            xml_parser.EndElementHandler = self._end_element

            try:
                xml_parser.Parse(xml_str)
            except xml.parsers.expat.ExpatError as errcode:
                self.result_dict['status'] = "XML parsing failed: " + str(errcode)
                self.result_dict['stdout'] = str(xmldata)
                xml_parse_ok = False

            if xml_parse_ok:
                self.result_dict['status'] = self.result_dict['xml_dict']['results']['attrs']['status']
                stdout_string = self._format_escaped_data(self.result_dict['xml_dict']['cli-output']['data'])
                self.result_dict['stdout'] = stdout_string
                for line in stdout_string.split('\n'):
                    stripped_line = line.strip()
                    if len(stripped_line) > 1:
                        self.result_dict['stdout_lines'].append(stripped_line)
                self.result_dict['xml_dict']['cli-output']['data'] = stdout_string
                self.result_dict['result_value'] = int(str(self.result_dict['xml_dict']['cli-result-value']['data']).replace("'", ""))

        return self.result_dict
```

Reply:
good for me